From f12685289eb3912191426212029c54eab204e24d Mon Sep 17 00:00:00 2001
From: Max Wipfli <max.wipfli@axelera.ai>
Date: Wed, 3 Jul 2024 15:08:15 +0000
Subject: [PATCH 3/9] Relocate MCU firmware to 0x28_0300_0000

This amounts to DDR_0_BASE + 48 MiB.

This change should not impact functionality, and is required for
integration into our address map.
---
 bin/mcu/Makefile                                   | 2 +-
 bin/mcu/src/mcu_hex2c.pl                           | 4 ++++
 sim/tests/g_custom/custom_axi_mcu/instructions.hex | 4 ++++
 sim/tests/g_custom/custom_mcu/instructions.hex     | 4 ++++
 4 files changed, 13 insertions(+), 1 deletion(-)

diff --git a/bin/mcu/Makefile b/bin/mcu/Makefile
index 916cc8a..a7ec79c 100644
--- a/bin/mcu/Makefile
+++ b/bin/mcu/Makefile
@@ -48,7 +48,7 @@ ${PROJECT}.bin	: ${PROJECT}.elf
 ifeq ($(TYPE),NEWMCU)
 sram_0000.MCU.mem	: ${PROJECT}.bin
 	echo //@000000000 // size: 30592 > $@
-	echo @000000080 // mcu >> $@
+	echo @140180080 // mcu >> $@
 	hexdump -v -e '1/4 "%08X\n"' $< | awk -v words=8 '{ v="";for(i=1;i<words;i++) {v="_"$$0v;getline;};v=$$0v;print v}' >> $@
 	for i in {1..256};do echo 00000000_00000000_00000000_00000000_00000000_00000000_00000000_00000000 >> $@; done
 else
diff --git a/bin/mcu/src/mcu_hex2c.pl b/bin/mcu/src/mcu_hex2c.pl
index bc06586..e861135 100755
--- a/bin/mcu/src/mcu_hex2c.pl
+++ b/bin/mcu/src/mcu_hex2c.pl
@@ -76,6 +76,10 @@ if($oldmcu) {
   print OUT "write,0.000,0x".(sprintf("%06X", $apbid+0x0420)).",0x00000000, // SET MCU APB END ADDR MSB\n";
   print OUT "write,0.000,0x".(sprintf("%06X", $apbid+0x0424)).",0xA3CFFFFF, // SET MCU APB END ADDR LSB\n";
 }
+print OUT "write,0.000,0x".(sprintf("%06X", $apbid+0x0450)).",0x00000028, // SET MCU INST OFFSET MSB\n";
+print OUT "write,0.000,0x".(sprintf("%06X", $apbid+0x0454)).",0x03000000, // SET MCU INST OFFSET LSB\n";
+print OUT "write,0.000,0x".(sprintf("%06X", $apbid+0x0458)).",0x00000028, // SET MCU DATA OFFSET MSB\n";
+print OUT "write,0.000,0x".(sprintf("%06X", $apbid+0x045C)).",0x03000000, // SET MCU DATA OFFSET LSB\n";
 print OUT "write,0.000,0x".(sprintf("%06X", $apbid+0x0428)).",0x00000000, // SET PERIPH ADDR MSB\n";
 print OUT "write,0.000,0x".(sprintf("%06X", $apbid+0x042C)).",0x08000000, // SET PERIPH ADDR LSB\n";
 print OUT "write,0.000,0x".(sprintf("%06X", $apbid+0x0438)).",0xFFFFFFFF, // SET MCU MTIMECMP\n";
diff --git a/sim/tests/g_custom/custom_axi_mcu/instructions.hex b/sim/tests/g_custom/custom_axi_mcu/instructions.hex
index 9b7e963..58e839c 100644
--- a/sim/tests/g_custom/custom_axi_mcu/instructions.hex
+++ b/sim/tests/g_custom/custom_axi_mcu/instructions.hex
@@ -7,6 +7,10 @@ write,0.000,0x080428,0x00000000, // SET PERIPH ADDR MSB
 write,0.000,0x08042C,0x08000000, // SET PERIPH ADDR LSB
 write,0.000,0x080438,0xFFFFFFFF, // SET MCU MTIMECMP
 write,0.000,0x08043C,0xFFFFFFFF, // SET MCU MTIMECMP
+write,0.000,0x080450,0x00000028, // SET MCU INST OFFSET MSB
+write,0.000,0x080454,0x03000000, // SET MCU INST OFFSET LSB
+write,0.000,0x080458,0x00000028, // SET MCU DATA OFFSET MSB
+write,0.000,0x08045C,0x03000000, // SET MCU DATA OFFSET LSB
 write,0.000,0x080400,0x1,        // START MCU clock
 irq,0.000,0x080018,0x1E,         // WAIT MCU2CPU IT
 end_frame,0,MCU,0,               // DUMP RESULT
diff --git a/sim/tests/g_custom/custom_mcu/instructions.hex b/sim/tests/g_custom/custom_mcu/instructions.hex
index 9dd0d0f..58ebd77 100644
--- a/sim/tests/g_custom/custom_mcu/instructions.hex
+++ b/sim/tests/g_custom/custom_mcu/instructions.hex
@@ -7,6 +7,10 @@ write,0.000,0x080428,0x00000000, // SET PERIPH ADDR MSB
 write,0.000,0x08042C,0x08000000, // SET PERIPH ADDR LSB
 write,0.000,0x080438,0xFFFFFFFF, // SET MCU MTIMECMP
 write,0.000,0x08043C,0xFFFFFFFF, // SET MCU MTIMECMP
+write,0.000,0x080450,0x00000028, // SET MCU INST OFFSET MSB
+write,0.000,0x080454,0x03000000, // SET MCU INST OFFSET LSB
+write,0.000,0x080458,0x00000028, // SET MCU DATA OFFSET MSB
+write,0.000,0x08045C,0x03000000, // SET MCU DATA OFFSET LSB
 write,0.000,0x080400,0x1,        // START MCU clock
 irq,0.000,0x080018,0x1E,         // WAIT MCU2CPU IT
 irq,0.000,0x080018,0x1E,         // WAIT MCU2CPU IT
-- 
2.43.0

